### PR TITLE
Make it possible to configure [testenv] extras

### PR DESCRIPTION
**Pull request by mgedmin**
Needed for https://github.com/zopefoundation/zope.testrunner/pull/116
where I want to use tox's factors to conditionally enable the 'subunit'
extra for tox environments like py38-subunit.

Apologies for the unrelated flake8 fixes that my editor highlighted to me.

### CODE REVIEW

File: config/pure-python/packages.txt [hunk 1]
```diff
@@ -95,3 +95,4 @@ z3c.pagelet
 zope.app.container
 zope.app.locales
 zope.app.zcmlfiles
+zope.testrunner
```
Comment (mgedmin) [inline config/pure-python/packages.txt:98]: oops this probably shouldn't be here yet

**Comment by icemac** [comment]
This breaks the idea to have additional `[testenv]` extras at the top of `testenv-additional`. See example use in https://github.com/zopefoundation/zope.app.locales/pull/13/files#diff-4425ec4892a0640706a7612e3c26160478a52d1cb052428124b2e79769ef72d2R17

I know it is not documented the way I used it. But I had it in mind when coding `testenv-additional` that it could be used for additional extras.
On the other hand I see the benefit of making it explicit.

**Comment by icemac** [comment]
Searching though GitHub it only seems to affect `zope.app.locales` where the PR is not yet merged but no packages which already have been merged, see https://github.com/search?q=org%3Azopefoundation+testenv-additional+language%3ATOML&type=Code&ref=advsearch&l=TOML&l=

File: config/README.rst [hunk 2]
```diff
@@ -319,6 +324,10 @@ additional-envlist
   ``testenv-additional`` (see below). This option has to be a list of strings
   without indentation.
 
+testenv-extras
+  Replacement for the default ``extras`` option in ``[testenv]`` of
```
Comment (icemac) [inline config/README.rst:328]: Does it actually have to be a replacement? Having `docs` automatically be added if `--with-sphinx-doctests` is used is actually quite handy.

Comment (mgedmin) [reply config/README.rst:328]: Eh, I mainly went with a replacement to keep the option name shorter.  I can create a PR for `testenv-additional-extras` if you prefer.

(I'm strongly against packages assuming `testenv-additional` can be used to extend an arbitrary option that happened to be right before the place %(testenv-additional)s got expanded at some point in time.)

Comment (icemac) [reply config/README.rst:328]: I'd prefer `testenv-additional-extras`.

It was probably not such a good idea to use `testenv-additional` for additional extras in the first place.

Comment (mgedmin) [reply config/README.rst:328]: I've done that in #76.

Comment (icemac) [reply config/README.rst:328]: Thank you!

**Pull request closed without merging**